app/controllers/main_controller.rb

```diff
@@ -60,7 +60,13 @@ def login
     session[:redirect_uri] = nil
     refresh_timeout
     # redirect to last visited page or to main page
-    redirect_to(uri || { controller: 'courses', action: 'index' })
+    if uri.present?
```

Comment:
@anoverga17 This is good, but I just noticed that you should make the same change for the code on lines 24-28 as well